src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -0,0 +1,201 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.maps.model.LatLng;
+import com.google.maps.errors.ApiException;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+/**
+ * Class that manages delivery slots.
+ */
+public class DeliverySlotManager {
+
+  /** 
+   * Class that creates a delivery slot.
+   */
+  private class CreateSlotManager {
+    private int timezoneOffset;
+    private DeliverySlot deliverySlot;
+    private int maxStops;
+    private double latitude;
+    private double longitude;
+
+    public CreateSlotManager(String deliveryDay, String timezoneOffsetMinutes, String startTime, String endTime, String maxStops) throws BadRequestException {
+      this.timezoneOffset = parseInteger(timezoneOffsetMinutes) * 60;
+      int startTimeSeconds = getNumberOfSeconds(parseTime(startTime)) + this.timezoneOffset;
+      int endTimeSeconds = getNumberOfSeconds(parseTime(endTime)) + this.timezoneOffset;
+      if (startTimeSeconds > endTimeSeconds) {
+        throw new BadRequestException("Please enter a valid time!");
+      }
+      this.deliverySlot = new DeliverySlot(validateDate(deliveryDay, startTimeSeconds), startTimeSeconds, endTimeSeconds);
+      this.maxStops = parseInteger(maxStops);
+      if (this.maxStops > 25 || this.maxStops <= 0) {
+        throw new BadRequestException("Please enter between 1 and 25 stops!");
+      }
+    }
+    
+    /** 
+     * Sets the coordinates for the starting point of the delivery.
+     * TODO[ak47na]: represent latitude and longitude as a Point object.
+     */
+    public void setCoordinates(String latitude, String longitude) throws BadRequestException {
+      this.latitude = parseDouble(latitude);
+      this.longitude = parseDouble(longitude);
+    }
+    
+    /**
+     * Calls Geocoding API to find the latitude and longitude of the provided address, and sets 
+     * them as the starting point of the delivery.
+     */
+    public void setCoordinatesFromAddress(String startAddress) throws ApiException, InterruptedException, IOException, DataNotFoundException {
+      LatLng point = MapsRequest.getLocationFromAddress(startAddress);
+      if (point == null) {
+        // Geocoding API didn't find the coordinates corresponding to given address.
+        throw new DataNotFoundException("Unable to find address!");
+      }
+      this.latitude = point.lat;
+      this.longitude = point.lng;
+    }
+
+    /**
+     * Return a Date object from dateString.
+     */
+    private Date validateDate(String dateString, int startTimeSeconds) throws BadRequestException {
+      SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
```

Comment:
Please can you put `"yyyy-MM-dd"` as a constant at the top of the file?

Makes it easier to find/update in the future, and makes future reuse easier.

Reply:
Right, done.